src/Sentry/Internal/IFileSystem.cs

```diff
@@ -1,14 +1,12 @@
 namespace Sentry.Internal;
 
-internal enum FileOperationResult
-{
-    Success,
-    Failure,
-    Disabled
-}
-
 internal interface IFileSystem
 {
+    // Note: You are responsible for handling success/failure when attempting to write to disk.
+    // You are required to check for `Options.FileWriteDisabled` whether you are allowed to call any writing operations.
```

Comment:
This is a good call out. Adding a sexton like <example> or <note> (can't recall the options) above methods in the interface with a snippet might make it even more obvious